core/src/main/scala/kafka/server/ConfigHelper.scala

```diff
@@ -93,6 +95,24 @@ class ConfigHelper(metadataCache: MetadataCache, config: KafkaConfig, configRepo
                 (name, value) => new DescribeConfigsResponseData.DescribeConfigsResourceResult().setName(name)
                   .setValue(value.toString).setConfigSource(ConfigSource.DYNAMIC_BROKER_LOGGER_CONFIG.id)
                   .setIsSensitive(false).setReadOnly(false).setSynonyms(List.empty.asJava))
+
+          case ConfigResource.Type.CLIENT_METRICS =>
+            val entityName = resource.resourceName
+            if (resource.resourceName == null || resource.resourceName.isEmpty) {
+              throw new InvalidRequestException("Client metrics subscription id must not be empty")
+            } else {
```

Comment:
you actually don't need the else here, as it will throw if the above condition is met.

Reply:
Agreed, I have just followed whatever the convention is used in the same method for other resource types.